uint8 axis
---
int32 shadow_count
int32 cpr_count